### PR TITLE
[JBPM-9747] Support springboot buildpack immutable images

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-9747

depends on https://github.com/kiegroup/kie-soup/pull/224/files

### CODE REVIEW

File: kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/autoconfiguration/KieServerAutoConfiguration.java [hunk 1]
```diff
@@ -76,7 +80,8 @@ public class KieServerAutoConfiguration extends AbstractJaxrsClassesScanServer {
     private String cxfPath;
 
     private static final Logger logger = LoggerFactory.getLogger(KieServerAutoConfiguration.class);
-    
+    private String EXPLODED_JAR_FOLDER = System.getProperty("org.kie.maven.resolver.folder");
```
Comment (gmunozfe) [inline kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/autoconfiguration/KieServerAutoConfiguration.java:83]: Probably it's better to use value annotation:
```suggestion
    @Value("${org.kie.maven.resolver.folder:/}")
    private String EXPLODED_JAR_FOLDER;
```

**Review by gmunozfe — CHANGES_REQUESTED** [review]
need to rebase and a comment related to the system property

File: kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/autoconfiguration/KieServerAutoConfiguration.java [hunk 2]
```diff
@@ -165,8 +172,32 @@ public List<KieContainerResource> buildDeployments(KieServerProperties kieServer
...
+
+    public List<KieContainerResource> discoverDeployments(String folder, File root) {
+        try {
+            Path base = Paths.get(root.toString(), folder);
+            if(!Files.isDirectory(base)) {
+                logger.error("Autoscan folder failed. {} does not exist or it is not a folder", base);
+                return Collections.emptyList();
+            }
+            List<KieContainerResource> files = new ArrayList<>();
+            List<Path> candidates = Files.walk(base).filter(Files::isRegularFile).collect(Collectors.toList());
```
Comment (gmunozfe) [inline kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/autoconfiguration/KieServerAutoConfiguration.java:186]: Use try-with-resources or close this "Stream" in a "finally" clause.

**Review by gmunozfe — CHANGES_REQUESTED** [review]
Files stream not closed

File: kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/autoconfiguration/KieServerAutoConfiguration.java (hunk 2 above)
Comment (elguardian) [reply kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration/src/main/java/org/kie/server/springboot/autoconfiguration/KieServerAutoConfiguration.java]: done

**Comment by elguardian** [comment]
Jenkins retest this please

**Pull request merged**